packages/blade/src/components/Input/_decisions/decisions.md

```diff
@@ -0,0 +1,295 @@
+| isLoading                                                                                         | `boolean`                                                                                                                                                                                                                                              | No       | `false`                                                   | Used to show a spinner when the the data is being loaded. **Discuss the position of the loader. Check options mentioned in the open questions below**                                                                                                                                 |
+| maxCharacters                                                                                     | `number`                                                                                                                                                                                                                                               | No       |                                                           | Used to show a character counter under the input field                                                                                                                                                                                                                                |
+| validationState                                                                                   | `error` , `success`                                                                                                                                                                                                                                    | No       |                                                           | Applies the style to the input field based on validation status and shows either `errorText` or `successText` respectively                                                                                                                                                            |
+| helpText                                                                                          | `string`                                                                                                                                                                                                                                               | No       |                                                           | Shown when we want to add some hint to the input field. Displayed under the input field. Only one of `helpText`, `errorText` or `successText` is shown at a time in the priority order as `errorText`, `successText`, `helpText`                                                      |
```

Comment:
**Suggestion:** Removing the help text in case there is an error or some other message visible, doesn't seem like a good experience. If user is entering invalid value then this help text would help the user in entering the correct value but with correct flow the error text would remove the help text from screen thus increasing the chances of invalid text more

Reply:
@kamleshchandnani few work-around I can think of at this time
 is that incase where helptext & errorText both are going configured,
 -  there help text can be made part of something like a under info symbol or a tooltip - _something which isn't visible all thetim_
 - help text in some way can be configured with error text like a very small cta with error text `show help` so that the user has the scope of seeing the help text even when input is in error state
 - Error text dismissable so that after dismissing the error message user can see the help text

Reply:
>* there help text can be made part of something like behind a info symbol or a tooltip - something which isn't visible all thetim
>* help text in some way can be configured with error text like a very small cta with error text show help so that the user has the scope of seeing the help text even when input is in error state

both the options will lead the user to perform an extra step for checking help text, we want it to be upfront.

>Error text dismissable so that after dismissing the error message user can see the help text
I guess this indeirectly will always exist, once the user has validated the actual text the help text will appear again.


also, what are the use cases where you think the error text will be out of context of help text? for eg:
If the helptext for password will be "Password must contain 8 characters" the error text will automatically be in that context already "Password should be of atleast 8 characters"

Reply:
We're going with assumption that error text & help text would be related, but that's not going to be the case most of the time.

Help text always doesn't contain the information about the validation, let's say someone is writing TAN Number, now help text could be telling the user which document one should check for looking for getting their TAN number. There could be an error that `TAN is not valid` - _this could be due to some static validation_, in this case, we removed the help text and made visible the error text. Here our above assumption doesn't fit